src/packages/piral-oidc/src/types.ts

```diff
@@ -1,10 +1,91 @@
-declare module 'piral-core/lib/types/custom' {
-  interface PiletCustomApi extends PiralOidcApi {}
+import 'piral-core';
+import { Profile } from 'oidc-client';
+
+/**
+ * Available configuration options for the OpenID Connect plugin.
+ */
+export interface OidcConfig {
+  /**
+   * The id of the client. Required for the setup of OAuth 2.0.
+   */
+  clientId: string;
+  /**
+   * The client secret.
+   */
+  clientSecret?: string;
+  /**
+   * The Uri pointing to the Identity Provider.
+   */
+  identityProviderUri: string;
+  /**
+   * The redirect Uri to use. By default the origin with /auth
+   * is used.
+   */
+  redirectUri?: string;
+  /**
+   * The Uri to which the Identity provider should redirect
+   * after a logout. By default the origin is used.
+   */
+  postLogoutRedirectUri?: string;
+  /**
+   * The protocol response type to be used. By default, `id_token`
+   * is used.
+   */
+  responseType?: string;
+  /**
+   * The scopes to be used. By default, `openid` is used.
+   */
+  scopes?: Array<string>;
+  /**
+   * Restricts token sharing such that other integrations, e.g., with
+   * fetch would need to be done manually.
+   * Otherwise, the client is responsive to the `before-fetch` event.
+   */
+  restrict?: boolean;
 }
 
-export interface PiralOidcApi {
+export type OidcProfileWithCustomClaims<TCustomClaims extends {}> = TCustomClaims & Profile;
+
+export interface OidcRequest {
+  /**
+   * Sets the headers of the request.
+   * @param headers Headers or a promise to headers.
+   */
+  setHeaders(headers: any): void;
+}
+
+export interface OidcClient<TCustomClaims extends {} = {}> {
+  /**
+   * Performs a login.
+   */
+  login(): void;
+  /**
+   * Performs a logout.
+   */
+  logout(): void;
+  /**
+   * Retrieves the current user profile.
+   */
+  profile(): Promise<OidcProfileWithCustomClaims<TCustomClaims>>;
```

Comment:
I would like to make a small suggestion here:
`piral-adal` calls this `account()`. While there *are* differences between ADAL's `Account` and oidc-client's `Profile`, I'd still consider naming the method `account()` in order to have a consistent public API. This should (a) make it easier for new users to get accustomed with Piral's auth plugins and (b) make it (somewhat) easier for long-term users to switch between different authentication flows.

@FlorianRappl

Reply:
Good point.

Well, at least we should discuss it. The resulting shape is definitely different between the different plugins, however, depending on the provider + options it may be different even within the same plugin (i.e., its conf. dependent).

Reply:
I think that's a fine suggestion, however I do think it may make sense to start namespacing things on the Piral API. It felt strange to me to place methods directly on the "global" instead of an object with the methods on it.

Example:
```ts
function setup(api) {
  api.getProfile()
}
```
vs
```ts
function setup(api) {
  api.oidc.getProfile()
  // or api.oidc.getAccount();
}
```

Reply:
You can certainly namespace things in your plugins - but for our plugins we do not plan to introduce this.

We want to avoid getting somewhere in `api.foo.bar.qxz.something()`, especially in cases where sub parts are suddenly not guaranteed to be available.

Note that the comment from @manuelroemer was not placed at the Pilet API, but at the `OidcClient`. The `getProfile` naming was totally fine. If you want to be more explicit you could name it `getOidcProfile` (or similar), too.

Reply:
Updated to rename `profile` to `account` on the piral-instance.